### PR TITLE
Make convert importable from python code

### PR DESCRIPTION
**Pull request by autra**
This is useful when calling py3dtiles from other python project.

We probably need a lot more flexibility than that (being able to add or remove workers on the fly for instance), and the api is still quite...rough (mimic cli too much imho), but this is a start.

### CODE REVIEW

**Comment by autra** [comment]
@delhomer or @oslandia for review please :-)

File: py3dtiles/convert.py [hunk 1]
```diff
@@ -290,35 +292,93 @@ def init_parser(subparser, str2bool):
...
     os.makedirs(working_dir)
 
     node_store = SharedNodeStore(working_dir)
 
     # read all input files headers and determine the aabb/spacing
-    _, ext = os.path.splitext(args.files[0])
+    _, ext = os.path.splitext(files[0])
     fn = las_reader.init if ext == '.las' else xyz_reader.init
-    infos = fn(args)
+    # TODO
```
Comment (delhomer) [inline py3dtiles/convert.py]: What does it mean? What are we supposed "to do"?

File: py3dtiles/convert.py [hunk 2]
```diff
@@ -290,35 +292,93 @@ def init_parser(subparser, str2bool):
...
+    working_dir = os.path.join(outfolder, '/tmp')
     os.makedirs(working_dir)
 
     node_store = SharedNodeStore(working_dir)
 
     # read all input files headers and determine the aabb/spacing
-    _, ext = os.path.splitext(args.files[0])
+    _, ext = os.path.splitext(files[0])
     fn = las_reader.init if ext == '.las' else xyz_reader.init
-    infos = fn(args)
```
Comment (delhomer) [inline py3dtiles/convert.py:312]: I suggest a more explicit function name, like `reader`, instead of `fn`.

**Review by delhomer — APPROVED** [review]
@autra ok for saying it is not enough, however it is a good first step. Maybe we could set it up as an issue, in order to continue this effort. Though OK for merging!

File: py3dtiles/convert.py (hunk 1 above)
Comment (autra) [reply py3dtiles/convert.py]: Nothing, it is done now :-) removing

File: py3dtiles/convert.py [hunk 3]
```diff
@@ -290,35 +292,93 @@ def init_parser(subparser, str2bool):
...
+            shutil.rmtree(outfolder)
         else:
-            print('Error, folder \'{}\' already exists'.format(folder))
+            print('Error, folder \'{}\' already exists'.format(outfolder))
             sys.exit(1)
 
-    os.makedirs(folder)
-    working_dir = folder + '/tmp'
+    os.makedirs(outfolder)
+    working_dir = os.path.join(outfolder, '/tmp')
```
Comment (autra) [inline py3dtiles/convert.py]: aaaanndd we missed that. TI `os.path.join('/foo','/bar')` is not *at all* equivalent to `os.path.join('/foo', 'bar')` :-/

File: py3dtiles/convert.py (hunk 2 above)
Comment (autra) [reply py3dtiles/convert.py:312]: I chose `init_reader_fn`

**Comment by autra** [comment]
Ok I'm testing, and if everything is working as expected, I'm merging. Btw, I think we need a test : CI should have caught my error about `os.path.join`.

File: py3dtiles/convert.py (hunk 3 above)
Comment (delhomer) [reply py3dtiles/convert.py]: You mean that you should drop the "/" from "/tmp" ? :)

Comment (autra) [reply py3dtiles/convert.py]: yep. I did it already.

**Comment by autra** [comment]
~~J'ai ajouté un test.~~ I have added a unit test :-)

**Pull request merged**